test/CMakeLists.txt

```diff
@@ -17,3 +17,9 @@ if(TARGET ${PROJECT_NAME}_utest)
   target_link_libraries(${PROJECT_NAME}_utest ${Boost_LIBRARIES} ${class_loader_LIBRARIES})
   add_dependencies(${PROJECT_NAME}_utest ${PROJECT_NAME}_TestPlugins1 ${PROJECT_NAME}_TestPlugins2)
 endif()
+
+catkin_add_gtest(${PROJECT_NAME}_unique_ptr_test unique_ptr_test.cpp)
+if(TARGET ${PROJECT_NAME}_unique_ptr_test)
```

Comment:
this test is never triggered because by default c++11 flag is not set for any rosdistro (c.f. http://build.ros.org/job/Kpr__class_loader__ubuntu_xenial_amd64/19/console#console-section-17)
Could you test if c++11 is available and set the flag only for this test? using this kind of logic:
```cmake
include(CheckCXXCompilerFlag)
CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
if(COMPILER_SUPPORTS_CXX11)
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
endif()
```
Thanks!